src/meta/src/stream/stream_manager.rs

```diff
@@ -263,80 +263,80 @@ impl GlobalStreamManager {
-                                            request_id,
-                                            actor_ids,
-                                        };
-                                        client.drop_actors(request).await
-                                    }
-                                });
-                                try_join_all(futures).await?;
-
-                                self.fragment_manager
-                                    .drop_table_fragments_vec(&HashSet::from_iter(std::iter::once(
-                                        table_id,
-                                    )))
-                                    .await?;
-                            }
-                            if !table_fragments.is_created() {
-                                tracing::debug!(
-                                    "cancelling streaming job {table_id} by issue cancel command."
-                                );
-                                self.barrier_scheduler
-                                    .run_command(Command::CancelStreamingJob(table_fragments))
-                                    .await?;
-                            }
-                            let _ = finish_tx.send(()).inspect_err(|_| {
-                                tracing::warn!("failed to notify cancelled: {table_id}")
+                            tracing::debug!("cancelling streaming job {table_id} in buffer queue.");
+                            let node_actors = table_fragments.worker_actor_ids();
+                            let cluster_info =
+                                self.cluster_manager.get_streaming_cluster_info().await;
+                            let node_actors = node_actors
+                                .into_iter()
+                                .map(|(id, actor_ids)| {
+                                    (
+                                        cluster_info.worker_nodes.get(&id).cloned().unwrap(),
+                                        actor_ids,
+                                    )
+                                })
+                                .collect_vec();
+                            let futures = node_actors.into_iter().map(|(node, actor_ids)| {
+                                let request_id = Uuid::new_v4().to_string();
+                                async move {
+                                    let client = self.env.stream_client_pool().get(&node).await?;
+                                    let request = DropActorsRequest {
+                                        request_id,
+                                        actor_ids,
+                                    };
+                                    client.drop_actors(request).await
+                                }
                             });
-                            return Err(MetaError::cancelled("create".into()));
+                            try_join_all(futures).await?;
```

Comment:
What if the try block returns here, will the job be deleted from the tracker?

Reply:
Oh yeah... `try` block will exit, but it does not delete the job. But it should. As such after this `try` block, we should still `delete_job`, in case it wasn't deleted in the `try` block.

Reply:
Good catch

Reply:
Resolved https://github.com/risingwavelabs/risingwave/pull/13070/commits/91914de947e23612f8a10ca87f112aaff9887efe